Include/cpython/listobject.h

```diff
@@ -21,6 +21,13 @@ typedef struct {
     Py_ssize_t allocated;
 } PyListObject;
 
+#ifdef Py_GIL_DISABLED
+typedef struct {
+    Py_ssize_t allocated;
+    PyObject *ob_item[1];
```

Comment:
Once this is out of the public header, let's use flexible array members like so:

```suggestion
    PyObject *ob_item[];
```

You'll need to adjust the sizing code that allocates the `_PyListArray`